### PR TITLE
#25 setup testing

### PR DESCRIPTION
**Pull request by VildeStabell**
Set up the possibility to write and run tests. Wrote tests for Player, Idea, Brain, Wall and EliminationPhase classes.

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
-sourceCompatibility = 1.7
+sourceCompatibility = 1.8
```
Comment (sigridge) [inline core/build.gradle:1]: This results in that we can't run the program in desktop (atleast not until we change something else)

File: core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        new EliminationPhase(new ArrayList<>(), maxSelected));
+        assertThrows("Should not allow null-brains",
+                IllegalArgumentException.class, () ->
+                        new EliminationPhase(new ArrayList<>(Arrays.asList(b1, null, b2)),
+                                maxSelected));
+    }
+
+    @Test
+    public void getBrains() {
+        assertEquals(Map.of(b1, false, b2, false, b3, false), ep1.getBrains());
```
Comment (sigridge) [inline core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java]: "of" does not work with my settings

File: core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    @Test
+    public void toggleBrain() {
+        assertThrows("Should not toggle brain if not in list",
+                IllegalArgumentException.class, () -> ep1.toggleBrain(new Brain()));
+        ep1.toggleBrain(b1);
+        assertEquals("Expected list with b1 but was: " + ep1.getSelectedBrains(),
+                new ArrayList<>(Collections.singletonList(b1)), ep1.getSelectedBrains());
+        ep1.toggleBrain(b2);
+        assertEquals("Expected list with b1 and b2 but was: " + ep1.getSelectedBrains(),
+                new ArrayList<>(Arrays.asList(b1, b2)), ep1.getSelectedBrains());
```
Comment (sigridge) [inline core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java]: This test fails for me, since it returns the brains in another order

**Review by sigridge — CHANGES_REQUESTED** [review]
I've added a few comments, and please put a new line at the end of each file that's missing a new line. (Easy to see in GitHub, as it shows a red circle at the end of each file thats missing this line)

File: core/build.gradle (hunk 1 above)
Comment (VildeStabell) [reply core/build.gradle:1]: It changes the language level from 7 to 8, allowing, among other things, the usage of lambda expressions. Did you click on the 'sync now' link that popped up for the gradle? That might fix the other issue you had as well

Comment (sigridge) [reply core/build.gradle:1]: Yes, I did. But running on andrioid worked fine, so we dont need to change it. You just asked to be informed if things didnt work on our computers

Comment (VildeStabell) [reply core/build.gradle:1]: Think I fixed it, please let me know if you still have problems with it

File: core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java (hunk 3 above)
Comment (VildeStabell) [reply core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java]: Should be fixed now ^^

File: core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java (hunk 2 above)
Comment (VildeStabell) [reply core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java]: That's weird, what error message are you getting? And what's your system's java version?

Comment (sigridge) [reply core/src/test/java/com/mygdx/game/models/EliminationPhaseTest.java]: "Cannot resolve method 'of' in 'Map'" and no clue

**Review by sigridge — APPROVED** [review]

**Pull request merged**